### DISCUSSION TITLE
fail to set monitor mode? (RTL8192EU)

### DISCUSSION THREAD

**Discussion by S2puppyS2** [aircrack-ng/aircrack-ng#2538] [Q&A]
i installed driver for my TL-wn823n adapter from https://github.com/clnhub/rtl8192eu-linux

airmon-ng returns this so i know the driver is successfully installed

PHY     Interface       Driver          Chipset

phy1    wlan0           rtl8192eu       TP-Link TL-WN823N v2/v3 [Realtek RTL8192EU]

'airmon-ng start wlan0' says (monitor mode enabled)
but when i  type iwconfig to check it is still in managed mode.

also 'airodump-ng start wlan0' returns

ioctl(SIOCSIWMODE) failed: Invalid argument
ioctl(SIOCSIWMODE) failed: Invalid argument
Error setting monitor mode on wlan0
Failed initializing wireless card(s): wlan0
                          

i also tried :

ifconfig wlan0 down
iwconfig wlan0 mode monitor

but it says:

Error for wireless request "Set Mode" (8B06) :
    SET failed on device wlan0 ; Operation not permitted.

fyi i had no problems with monitor mode before installing the driver
but i started having problem with aireplay-ng sending deauth packets so i installed the driver

i am a complete beginner so i have no idea what i need to do
any help would be greatly appreciated

**Comment by ZerBea** [comment]
Some of newer Realtek drivers do not use WIRELESS EXTENSIONS any longer and ioctl() system call will fail.
iwconfig is a WIRELESS EXTENSION dinosaur that set monitor mode by WIRELESS EXTENSIONS. This might fail in that case.
Here is an overview of deprecated tools and their replacements:
https://www.tecmint.com/deprecated-linux-networking-commands-and-their-replacements/  
iwconfig is among them and replaced by iw

The Linux stock kernel driver support this interface, so it is definitely not a good idea to replace it by a third party driver.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/net/wireless/realtek/rtl8xxxu?h=v6.2.101

Additional, there are several unfixed problems on the third party driver:
https://github.com/clnhub/rtl8192eu-linux/issues

By installing a third party driver, you now you have problems with the driver and with aireplay-ng.

Here are two solutions to get more information about the capabilities of driver and interface:
first - use iw to get information about the driver:
$ iw dev
$ iw list

second - use hcxdumptool to get information  about the capabilities of driver and interface:
$ hcxdumptool -L
$ hcxdumptool -I wlan0

I neither use iw nor airmon-ng so here is an example output of latest hcxdumptool git head and the Linux stock kernel driver supplied by the kernel
$ uname -r
6.2.10-arch1-1

```
$ hcxdumptool -L

Requesting interface capabilities. This may take some time.
Please be patient...

available wlan devices:

phy idx hw-mac       virtual-mac  m ifname           driver (protocol)
---------------------------------------------------------------------------------------------
 28  30 f81a670dc527 ea2e9f00757c + wlp5s0f4u2       rtl8xxxu (NETLINK & WIRELESS EXTENSIONS)

* active monitor mode available
+ monitor mode available
- no monitor mode available

bye-bye
```

```
$ hcxdumptool -I wlp5s0f4u2

Requesting interface capabilities. This may take some time.
Please be patient...

interface information:

phy idx hw-mac       virtual-mac  m ifname           driver (protocol)
---------------------------------------------------------------------------------------------
 28  30 f81a670dc527 ea2e9f00757c + wlp5s0f4u2       rtl8xxxu (NETLINK & WIRELESS EXTENSIONS)

available frequencies: frequency [channel] tx-power of Regulatory Domain: 98

  2412 [  1] 20.0 dBm	  2417 [  2] 20.0 dBm	  2422 [  3] 20.0 dBm	  2427 [  4] 20.0 dBm
  2432 [  5] 20.0 dBm	  2437 [  6] 20.0 dBm	  2442 [  7] 20.0 dBm	  2447 [  8] 20.0 dBm
  2452 [  9] 20.0 dBm	  2457 [ 10] 20.0 dBm	  2462 [ 11] 20.0 dBm	  2467 [ 12] 20.0 dBm
  2472 [ 13] 20.0 dBm	  2484 [ 14] disabled

bye-bye
```


BTW:
Setting monitor mode is working as expected running the Linux stock kernel driver:
```
$ sudo hcxdumptool -m wlp5s0f4u2

Requesting interface capabilities. This may take some time.
Please be patient...

interface information:

phy idx hw-mac       virtual-mac  m ifname           driver (protocol)
---------------------------------------------------------------------------------------------
 29  31 f81a670dc527 06afab6ec1ed + wlp5s0f4u2       rtl8xxxu (NETLINK & WIRELESS EXTENSIONS)

available frequencies: frequency [channel] tx-power of Regulatory Domain: 98

  2412 [  1] 20.0 dBm	  2417 [  2] 20.0 dBm	  2422 [  3] 20.0 dBm	  2427 [  4] 20.0 dBm
  2432 [  5] 20.0 dBm	  2437 [  6] 20.0 dBm	  2442 [  7] 20.0 dBm	  2447 [  8] 20.0 dBm
  2452 [  9] 20.0 dBm	  2457 [ 10] 20.0 dBm	  2462 [ 11] 20.0 dBm	  2467 [ 12] 20.0 dBm
  2472 [ 13] 20.0 dBm	  2484 [ 14] disabled


monitor mode is active...

bye-bye
```

**Reply by S2puppyS2** [reply]
thank you, so should i remove the installed driver and enable the rtl8xxxu again? how do i uninstall the drivers and get the rtl8xxxu driver up and running?? i am using kali linux btw

**Reply by ZerBea** [reply]
"Should I remove the third party driver?"
Yes, if it doesn't support monitor mode and packet injection

"How do i uninstall the drivers and get the rtl8xxxu driver up and running?"
Read the documentation of the third party drive.

"I am using KALI Linux."
Definitely not a distribution for a beginner.

**Comment by ZerBea** [comment]
BTW:
We discovered a problem on rtl8xxxu that can cause that the driver does not transmit packets in monitor mode (rtl8818eu). The hunt on this bug has started and you can watch progress, here:
https://bugzilla.kernel.org/show_bug.cgi?id=217205

Additional I opened a feature request to add active monitor mode:
https://bugzilla.kernel.org/show_bug.cgi?id=217319

**Comment by ZerBea** [comment]
BTW:
You commented:
```
fyi i had no problems with monitor mode before installing the driver
but i started having problem with aireplay-ng sending deauth packets so i installed the driver
```
Injecting DEAUTHENTICATION frames is not the best solution:
https://github.com/ZerBea/hcxtools/issues/265#issuecomment-1501589882
looks like aireplay-ng injecting DEAUTHENTICATION frames directly into AUTHENTICATION sequences.
Neither the AP nor the CLIENT have a chance to complete the AUTHENTICATION sequence.

**Reply by S2puppyS2** [reply]
the weird thing is, the very first time i used aireplay-ng to send deauth packets it worked fine (with the rtl8xxxu driver), i could even see my device disconnected from the router  but today it would send maybe 3 or 4 packets and then just stop even though thats way less than the amount i specified. so i thought there was something wrong with the driver and decided to install the rtl8192eu.

**Reply by ZerBea** [reply]
I have no problems (except the calibration problem mentioned above) on the stock kernel driver on kernel 6.2.10.
